packages/terra-form-checkbox/src/Checkbox.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Checkbox.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+    * The checked property of the Input element. Use this to generate a controlled Control Element.
+    */
+  checked: PropTypes.bool,
+  /**
+    * The checked property of the Input element. Use this to generate an uncontrolled Control Element.
+    */
+  defaultChecked: PropTypes.bool,
+  /**
+    * The id of the input field.
+    */
+  id: PropTypes.string,
+  /**
+    * Additional attributes for the input object.
+    */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the control element is disabled or not.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+    * Whether the control element is inline or not.
+    */
+  isInline: PropTypes.bool,
+  /**
+    * Whether the label is hidden or not.
+    */
+  isLabelHidden: PropTypes.bool,
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+  /**
+    * Additional objects for the text object.
+    */
+  labelTextAttrs: PropTypes.object,
+  /**
+    * Name of the input attribute.
+    */
+  name: PropTypes.string,
+  /**
+    * Function to trigger when user clicks on the input. Provide a function to create a controlled input.
+    */
+  onChange: PropTypes.func,
+  /**
+    * The value of the input element.
+    */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  checked: undefined,
+  defaultChecked: false,
+  id: undefined,
+  inputAttrs: {},
+  isDisabled: false,
+  isInline: false,
+  isLabelHidden: false,
+  labelText: null,
+  labelTextAttrs: {},
+  name: null,
+  onChange: undefined,
+  value: undefined,
+};
+
+class Checkbox extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = { focus: false };
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+    this.getFocusClass = this.getFocusClass.bind(this);
+  }
+
+  onFocus() {
+    this.setState({ focus: true });
```

Comment:
I'm trying to think if there's a better way to accomplish your focus styling than having focus as state. But if you keep it this way you'll need to call the onFocus and onBlur functions passed in through the inputAttrs props, if they exist.